### PR TITLE
Adding hipMemset functional tests

### PR DESCRIPTION
**Pull request by dylanangus**
Added functional tests for hipMemset API (hipMemset, hipMemsetD8, hipMemsetD16, hipMemsetD32, , hipMemset2D, , hipMemset3D) and async equivalents.

This PR depends on other PRs for the HIP_CHECK_ERROR macro and will be rebased once merged.

### CODE REVIEW

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -51,17 +51,19 @@ def hipBuildTest(String backendLabel) {
...
-                sh  """#!/usr/bin/env bash
-                    set -x
-                    # Check if backend label contains string "amd" or backend host is a server with amd gpu
-                    if [[ $backendLabel =~ amd ]]; then
-                        LLVM_PATH=/opt/rocm/llvm ctest -E 'cooperative_streams_least_capacity.tst|cooperative_streams_half_capacity.tst|cooperative_streams_full_capacity.tst|grid_group_data_sharing.tst'
-                    else
-                        make test
-                    fi
-                    """
+        timeout(time: 1, unit: 'HOURS') {
```
Comment (chrispaquot) [inline .jenkins/Jenkinsfile]: what's all that change about?

**Comment by fabiomestre** [comment]
This PR is not ready to be merged

**Comment by fabiomestre** [comment]
This PR is now ready to be merged

**Review by chrispaquot — APPROVED** [review]

File: .jenkins/Jenkinsfile (hunk 1 above)
Comment (fabiomestre) [reply .jenkins/Jenkinsfile]: The original PR, included some extra commits from develop unintentionally. I spent some time a few weeks ago reverting changes that were not related to this PR. Including the one you just mentioned. If you check "files changed" there is only 2 files being updated at the moment: https://github.com/ROCm-Developer-Tools/HIP/pull/2669/files

I think if the PR is squashed when merging then the other commits will disappear. However, I understand that this is a bit messy. So I could create a new PR to fix it

Comment (fabiomestre) [reply .jenkins/Jenkinsfile]: Here is a new PR that doesn't contain any commits related to Jenkins: https://github.com/ROCm-Developer-Tools/HIP/pull/2791

Feel free to merge this one instead

**Pull request merged**